src/d_main.c

```diff
@@ -1741,6 +1741,30 @@ void D_SetBloodColor(void)
   }
 }
 
+void D_SetTranslucency(void)
+{
+  int i;
+
+  const int predefined_translucency[] = {
+    // MBF
+    MT_FIRE,      MT_SMOKE,     MT_FATSHOT,  MT_BRUISERSHOT,
+    MT_SPAWNFIRE, MT_TROOPSHOT, MT_HEADSHOT, MT_PLASMA,
+    MT_BFG,       MT_ARACHPLAZ, MT_PUFF,     MT_TFOG,
+    MT_IFOG,      MT_MISC12,    MT_INV,      MT_INS,
+    MT_MEGA,
+    // [Woof!]
+    MT_PLASMA1,   MT_PLASMA2
+  };
+
+  for (i = 0; i < arrlen(predefined_translucency); ++i)
+  {
+    if (general_translucency || (strictmode && !demo_compatibility))
```

Comment:
There is one thing that PrBoom+ does better here: it keeps track of which mobjinfos had their flags changed by DEHACKED and leaves them unchanged.